components/core/tests/test-ffi_SchemaTree.cpp

```diff
@@ -0,0 +1,107 @@
+#include <optional>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+#include <msgpack.hpp>
+
+#include "../src/clp/ffi/SchemaTree.hpp"
+#include "../src/clp/ffi/SchemaTreeNode.hpp"
+
+using clp::ffi::SchemaTree;
+using clp::ffi::SchemaTreeNode;
+
+namespace {
+/**
+ * @param schema_tree
+ * @param locator
+ * @param expected_id
+ * @return Whether the node is inserted successfully with the expected node id.
+ */
+[[nodiscard]] auto insert_node(
+        SchemaTree& schema_tree,
+        SchemaTree::NodeLocator locator,
+        SchemaTreeNode::id_t expected_id
+) -> bool;
+
+/**
+ * @param schema_tree
+ * @param locator
+ * @param expected_id
+ * @return Whether the node exists and its ID matches the expected ID.
+ */
+[[nodiscard]] auto check_node(
+        SchemaTree const& schema_tree,
+        SchemaTree::NodeLocator locator,
+        SchemaTreeNode::id_t expected_id
+) -> bool;
+
+auto insert_node(
+        SchemaTree& schema_tree,
+        SchemaTree::NodeLocator locator,
+        SchemaTreeNode::id_t expected_id
+) -> bool {
+    return false == schema_tree.has_node(locator)
+           && expected_id == schema_tree.insert_node(locator);
+}
+
+auto check_node(
+        SchemaTree const& schema_tree,
+        SchemaTree::NodeLocator locator,
+        SchemaTreeNode::id_t expected_id
+) -> bool {
+    auto const node_id{schema_tree.try_get_node_id(locator)};
+    return node_id.has_value() && node_id.value() == expected_id;
+}
+}  // namespace
+
+TEST_CASE("ffi_schema_tree", "[ffi]") {
+    /**
+     * <0:root:Obj>
```

Comment:
Should we use `0::Obj` since the root doesn't have a name?

Reply:
I will probably leave it so it's more explicit that it's the root node